### ISSUE TITLE
only prevent breaks occurring in the google_calendar plugin when you are actually participating in the event

### ISSUE THREAD

**Issue by cdepillabout** [cdepillabout/break-time#13]
Google Calendar has a way to mark that you are participating or not participating in an event that you have been invited to.

The Rust [google calendar library](https://docs.rs/google-calendar3/1.0.10+20190702/google_calendar3) exposes this info an `EventAttendee` type: https://docs.rs/google-calendar3/1.0.10+20190702/google_calendar3/struct.EventAttendee.html

This is in the `response_status` field: https://docs.rs/google-calendar3/1.0.10+20190702/google_calendar3/struct.EventAttendee.html#structfield.response_status

Ideally, break-time would not stop a break occurring for an event when your `response_status` is marked as `declined`.

Here is the function that looks whether or not each calendar has events right as a break is about to occur:

https://github.com/cdepillabout/break-time/blob/e2f26345539c146615f03f41dde0a5915c3d02df/src/scheduler/plugins/google_calendar.rs#L331-L341

This function would have to be updated to look at the `response_status` of everyone in the event (where the email address matches the email address of the given google calendar).

**Comment by cdepillabout** [comment]
Similarly, Google Calendar has a way to mark yourself as "Out of office".

If you have marked yourself as out of the office, then breaks should occur as normal.

When you mark yourself as "Out of office", the Google Calendar `Event` has a `summary` field of `Some("Out of office")`.

**Comment by cdepillabout** [comment]
I'm seeing a weird thing where even though I have declined an event, I get the event status as:

```
 EventAttendee
                { comment: None
                , display_name: None
                , self_: Some ( true )
                , email: Some ( "me@example.com" )
                , id: None
                , additional_guests: None
                , resource: None
                , organizer: None
                , optional: None
                , response_status: Some ( "needsAction" )
                }
```

I think this needs to be updated so that events are ignored even with a response_status of `needsAction`.

**Comment by cdepillabout** [comment]
Fixed the `needsAction` problem in #22.
